src/main/scala/com/banzaicloud/spark/metrics/sink/PrometheusSink.scala

```diff
@@ -221,20 +195,20 @@ abstract class PrometheusSink(property: Properties, registry: MetricRegistry)  e
   logInfo(s"$KEY_PUSHGATEWAY_ADDRESS_PROTOCOL -> $pushGatewayAddressProtocol")
   logInfo(s"$KEY_METRICS_NAME_CAPTURE_REGEX -> ${metricsNameCaptureRegex.getOrElse("")}")
   logInfo(s"$KEY_METRICS_NAME_REPLACEMENT -> $metricsNameReplacement")
-  logInfo(s"$KEY_LABELS -> ${labelsMap.getOrElse("")}")
+  logInfo(s"$KEY_LABELS -> ${labelsMap}")
   logInfo(s"$KEY_JMX_COLLECTOR_CONFIG -> $jmxCollectorConfig")
 
-  val pushRegistry: CollectorRegistry = CollectorRegistry.defaultRegistry
+  val pushRegistry: CollectorRegistry = new DeduplicatedCollectorRegistry()
 
-  lazy val sparkMetricExports: DropwizardExports = metricsNameCaptureRegex match {
-    case Some(r) => new DropwizardExportsWithMetricNameCaptureAndReplace(r, metricsNameReplacement, registry)
-    case _ => new com.banzaicloud.spark.metrics.DropwizardExports(registry)
-  }
+  private val pushTimestamp = if (enableTimestamp) Some(PushTimestampProvider()) else None
```

Comment:
This will lead to all metrics samples to have different timestamp when `enableTimestamp` is set to true as  `PushTimestampProvider` returns the current timestamp whenever  `provider.GetTimestamp()` is invoked. When `enableTimestamp` is set to true each metrics samples pushed to Pushgateway at once should have the same time stamp which is the time of the push.

Reply:
@stoader timestamp provider always return a new timestamp. However in the latest commit I changed the code to use it only once per collect method https://github.com/banzaicloud/spark-metrics/pull/46/files/4dd871a8f18ba63b3e5bc2b5e707370450c76091#diff-b3ab4b11ec7ec0a1186a0ebae74067deR70